Runtime/Scripts/PersistentConnection.cs

```diff
@@ -296,104 +327,176 @@ IEnumerator OnIntervalEdgeEventsLatency(EdgeEventsConnection connection, string
+              PropagateError(FindCloudletEventTrigger.CloudletMaintenanceStateChanged, "Received CloudletStateChanged Event but NewCloudlet is null");
+              return;
             }
-            return;
-        }
+            PropagateSuccess(FindCloudletEventTrigger.CloudletStateChanged, edgeEvent.NewCloudlet);
+          }
+          return;
+        case ServerEventType.EventCloudletState:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.CloudletStateChanged))
+          {
+            if (edgeEvent.NewCloudlet == null)
+            {
+              PropagateError(FindCloudletEventTrigger.CloudletStateChanged, "Received CloudletStateChanged Event but NewCloudlet is null");
+              return;
+            }
+            PropagateSuccess(FindCloudletEventTrigger.CloudletStateChanged, edgeEvent.NewCloudlet);
+          }
+          return;
+        case ServerEventType.EventCloudletUpdate:
+          if (fcTriggers.Contains(FindCloudletEventTrigger.CloserCloudlet))
+          {
+            if (edgeEvent.NewCloudlet == null)
+            {
+              PropagateError(FindCloudletEventTrigger.CloserCloudlet, "Received CloserCloudlet Event but NewCloudlet is null");
+              return;
+            }
+            PropagateSuccess(FindCloudletEventTrigger.CloserCloudlet, edgeEvent.NewCloudlet);
+          }
+          return;
+        case ServerEventType.EventError:
+          Logger.Log("Received EventError from server, " + edgeEvent.ToString());
+          PropagateError(FindCloudletEventTrigger.Error, edgeEvent.ErrorMsg);
+          return;
+        default:
+        case ServerEventType.EventUnknown:
+          Logger.Log("Received Unknown event, "+edgeEvent.ToString());
+          return;
       }
     }
 
-    async Task RespondToLatencyRequest()
+    IEnumerator RespondToLatencyRequest()
     {
       string host = integration.GetHost();
-      if (!LocationUpdate())
-      {
-        return;
-      }
+      yield return StartCoroutine(UpdateLocation());
       if (hasTCPPorts)
       {
         Logger.Log("Sending Latency Samples (TCP) as a response to LatencyRequest from the server");
-        await integration.matchingEngine.EdgeEventsConnection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location);
+        integration.matchingEngine.EdgeEventsConnection.TestConnectAndPostLatencyUpdate(host, (uint)config.latencyTestPort, location).ConfigureAwait(false);
       }
       else
       {
         Logger.Log("Sending Latency Samples (UDP) as a response to LatencyRequest from the server");
-        await integration.matchingEngine.EdgeEventsConnection.TestPingAndPostLatencyUpdate(host, location);
+        integration.matchingEngine.EdgeEventsConnection.TestPingAndPostLatencyUpdate(host, location).ConfigureAwait(false);
       }
     }
 
-    async void ProcessLatency(Statistics stats)
+    async Task ProcessLatency(Statistics stats)
```

Comment:
Mini Nit: Maybe a tiny Util wrapper that converts Task to IEnumerator for Unity users (while not complete, yield break).

Reply:
Why?!
This is not intended for Unity devs to use. It's a private function.